src/coffea/nanoevents/methods/nanoaod.py

```diff
@@ -506,6 +506,10 @@ class Jet(vector.PtEtaPhiMLorentzVector, base.NanoCollection, base.Systematic):
     TIGHTLEPVETO = 2
     "jetId bit position"
 
+    @property
+    def charge(self):
+        return 0.0 * self.pt
```

Comment:
can you try using `ak.zeros_like(self.pt)`, it will read less data. There are a few other places in the file that could use this optimization as well, please fix them.

Reply:
Does the dtype matter btw? Is there a reason it's `0.0 * self.pt` to return a float specifically. If yes, it should be set in `zeros_like` as well.

Reply:
zeros_like respects the dtype of the input.

Reply:
I have made the relevant changes to the nanoaod.py. I notice that `0.0 * self.pt` code is also present in delphs.py. Is there I way I could add delphs.py to this pull request? If not, I can always just make a new pull request

Reply:
ok NVM, the pull request automatically does it for you once you commit to the same branch